### PR TITLE
feat(planning_validator): add validation for trajectory length and related tests

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

This PR improves planning validation.

- Two validations are added: 
  - check trajectory forward length 
  - check longitudinal deviation
- Add tests for all validation in the ros node level
- Fix some bugs, and missing codes. (These are commented in github)

Now, the planning_validator can detect the invalid planning trajectory more correctly. For example, a very short path can be detected.

**Before**

https://github.com/autowarefoundation/autoware.universe/assets/21360593/914e75d2-db19-481d-abdb-be097e144e87

**After**

https://github.com/autowarefoundation/autoware.universe/assets/21360593/29e89df3-5183-4b0d-ab03-2ddf79bc1c6a

## Related links

None

## Tests performed

- run test in colcon
- run psim and check if undesired behavior occurs (no error message on terminal)

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

The planning validation may will make more errors in diagnostics.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/planning_validator/src/utils.cpp [hunk 1]
```diff
@@ -296,7 +296,7 @@ std::pair<double, size_t> calcMaxSteeringRates(
     const auto steer_next = steering_array.at(i + 1);
 
     const auto steer_rate = (steer_next - steer_prev) / dt;
-    takeBigger(max_steering_rate, max_index, steer_rate, i);
+    takeBigger(max_steering_rate, max_index, std::abs(steer_rate), i);
```
Comment (TakaHoribe) [inline planning/planning_validator/src/utils.cpp:299]: Fix bug.

File: planning/planning_validator/src/planning_validator.cpp [hunk 2]
```diff
@@ -148,6 +152,20 @@ void PlanningValidator::setupDiag()
     setStatus(
       stat, validation_status_.is_valid_velocity_deviation, "velocity deviation is too large");
   });
+  d->add(ns + "distance_deviation", [&](auto & stat) {
```
Comment (TakaHoribe) [inline planning/planning_validator/src/planning_validator.cpp:155]: Add missing diagnostic

File: planning/planning_validator/test/src/test_planning_validator_pubsub.cpp [hunk 3]
```diff
@@ -55,7 +59,7 @@ class PubSubManager : public rclcpp::Node
 
 void spinSome(rclcpp::Node::SharedPtr node_ptr)
 {
-  for (int i = 0; i < 50; ++i) {
+  for (int i = 0; i < 10; ++i) {
```
Comment (TakaHoribe) [inline planning/planning_validator/test/src/test_planning_validator_pubsub.cpp:62]: Need to make sure this spin time is enough for CICD. This works well in my environment.

File: map/map_projection_loader/src/map_projection_loader.cpp [hunk 4]
```diff
@@ -95,4 +95,5 @@ MapProjectionLoader::MapProjectionLoader() : Node("map_projection_loader")
   const auto adaptor = component_interface_utils::NodeAdaptor(this);
   adaptor.init_pub(publisher_);
   publisher_->publish(msg);
+  RCLCPP_WARN(get_logger(), "published map projection message");
```
[the remark this answers is not in the document]
Comment (TakaHoribe) [reply map/map_projection_loader/src/map_projection_loader.cpp:98]: @SakodaShintaro Sorry, this is just my mistake. This PR won't have any modification out of the planning_validator pkg.

File: planning/planning_validator/src/planning_validator.cpp [hunk 5]
```diff
@@ -447,13 +472,84 @@ bool PlanningValidator::checkValidDistanceDeviation(const Trajectory & trajector
...
+
+  const auto forward_length = motion_utils::calcSignedArcLength(
+    trajectory.points, current_kinematics_->pose.pose.position, trajectory.points.size() - 1);
+
+  const auto acc = validation_params_.forward_trajectory_length_acceleration;
+  const auto forward_length_required = ego_speed * ego_speed / (2.0 * std::abs(acc)) -
+                                       validation_params_.forward_trajectory_length_margin;
+
+  std::cerr << "forward_length_required = " << forward_length_required
+            << ", forward_length = " << forward_length << std::endl;
```
Comment (kosuke55) [inline planning/planning_validator/src/planning_validator.cpp:537]: is this needed?

**Review by kosuke55 — APPROVED** [review]
LGTM!

**Pull request merged**